debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -93,6 +93,10 @@ public void execute(ChangeEventSourceContext context) throws InterruptedExceptio
             LOGGER.info("Last position recorded in offsets is {}", lastProcessedPositionOnStart);
 
             TxLogPosition lastProcessedPosition = lastProcessedPositionOnStart;
+
+            // LSN should be increased for the first run only immediately after snapshot completion
+            // otherwise we might skip an incomplete transaction after restart
+            boolean shouldIncreaseFromLsn = offsetContext.isSnapshotCompleted();
```

Comment:
Just to be sure - what is the value of snapshotCompleted when the offset saved is like `{"commit_lsn":"000443d8:0073d0da:0005","change_lsn":"000443d8:0073d037:00a8"}`?

Reply:
@jpechane Yes, it is false. I've added two more test case to cover potential paths.